### PR TITLE
Resolving #868

### LINKED ISSUE #868: Audit system/hal/core for console output
Low level drivers / modules as VFSManager should never write using Console.Write / Console.WriteLine but at least an example of this can be seen here: https://github.com/CosmosOS/Cosmos/blob/master/source/Cosmos.System2/FileSystem/VFS/VFSManager.cs#L278

if debug logs are needed, we have the debugger for that.

### PR DESCRIPTION
**Pull request by KrzysztofSzewczyk**
Resolve #868, allow user kernel to handle exceptions.
This is my first contribution, if something is wrong please tell me and optionally help me to fix.

### CODE REVIEW

File: source/Cosmos.System2_Plugs/System/IO/PathImpl.cs [hunk 1]
```diff
@@ -401,8 +401,8 @@ public static string GetDirectoryName(string aPath)
                 string xText = NormalizePath(aPath, false);
                 if (aPath.Length > 0)
                 {
-                    try
```
Comment (charlesbetros) [inline source/Cosmos.System2_Plugs/System/IO/PathImpl.cs]: This should stay. It came directly from .net sources.

**Comment by KrzysztofSzewczyk** [comment]
@charlesbetros  Done1

**Comment by fanoI** [comment]
As you can see here https://ci.appveyor.com/project/CosmosOS/cosmos/build/0.20150918.1724#L3590 it is not compiling.
Try to add:
using Sys = Cosmos.System;

and then change Global.mDebugger.SendInterna() to Sys.Global.mDebugger.SendInternal()

**Comment by charlesbetros** [comment]
This PR hasn't had any updates in a while. Most files are out of date or in conflict. Closing this, though feel free to resubmit with updated sources.

**Pull request closed without merging**